common/Storyboarding/OsbSpriteWriter.cs

```diff
@@ -0,0 +1,306 @@
+                }
+
+                segment.Add(command);
+            }
+
+            AddStaticCommands(segment, startTime);
+
+            fragmentationTimes.RemoveWhere(t => t < endTime);
+            commands.RemoveAll(c => c.EndTime <= endTime);
+
+            return segment;
+        }
+
+        private void AddStaticCommands(List<ICommand> segment, int startTime)
+        {
+            if (moveTimeline.HasCommands && !segment.Any(c => c is MoveCommand && c.StartTime == startTime))
+            {
+                var value = moveTimeline.ValueAtTime(startTime);
+                segment.Add(new MoveCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+
+            if (moveXTimeline.HasCommands && !segment.Any(c => c is MoveXCommand && c.StartTime == startTime))
+            {
+                var value = moveXTimeline.ValueAtTime(startTime);
+                segment.Add(new MoveXCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+
+            if (moveYTimeline.HasCommands && !segment.Any(c => c is MoveYCommand && c.StartTime == startTime))
+            {
+                var value = moveYTimeline.ValueAtTime(startTime);
+                segment.Add(new MoveYCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+
+            if (rotateTimeline.HasCommands && !segment.Any(c => c is RotateCommand && c.StartTime == startTime))
+            {
+                var value = rotateTimeline.ValueAtTime(startTime);
+                segment.Add(new RotateCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+
+            if (scaleTimeline.HasCommands && !segment.Any(c => c is ScaleCommand && c.StartTime == startTime))
+            {
+                var value = scaleTimeline.ValueAtTime(startTime);
+                segment.Add(new ScaleCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+
+            if (scaleVecTimeline.HasCommands && !segment.Any(c => c is VScaleCommand && c.StartTime == startTime))
+            {
+                var value = scaleVecTimeline.ValueAtTime(startTime);
+                segment.Add(new VScaleCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+
+            if (colorTimeline.HasCommands && !segment.Any(c => c is ColorCommand && c.StartTime == startTime))
+            {
+                var value = colorTimeline.ValueAtTime(startTime);
+                segment.Add(new ColorCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+
+            if (fadeTimeline.HasCommands && !segment.Any(c => c is FadeCommand && c.StartTime == startTime))
+            {
+                var value = fadeTimeline.ValueAtTime(startTime);
+                segment.Add(new FadeCommand(OsbEasing.None, startTime, startTime, value, value));
+            }
+        } 
+    }
+
+    static class OsbSpriteExtensions

```

Comment:
No need for extensions, move these as methods of this class

Reply:
done

Reply:
it's still here?

Reply:
done for real now